test/infrastructure/docker/internal/controllers/dockermachine_controller.go

```diff
@@ -119,6 +120,10 @@ func (r *DockerMachineReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, nil
 	}
 
+	if isPaused, conditionChanged, err := paused.EnsurePausedCondition(ctx, r.Client, cluster, dockerMachine); err != nil || isPaused || conditionChanged {
+		return ctrl.Result{}, err
+	}
+
 	log = log.WithValues("Cluster", klog.KObj(cluster))
 	ctx = ctrl.LoggerInto(ctx, log)
```

Comment:
Let's drop the annotations.IsPaused below. Can you please double-check all usages of this and similar funcs that we got rid of all of them were it made sense? (mostly in controllers like here I guess)

Also let's move this below the part where we add the Cluster k/v pair (please check if there are similar cases in other controllers)